src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -220,9 +219,12 @@ await Parallel.ForEachAsync(resources, parallelOptions, async (resource, innerCt
 
         public async Task<UpsertOutcome> UpsertAsync(ResourceWrapperOperation resource, CancellationToken cancellationToken)
         {
-            bool isBundleOperation = _bundleOrchestrator.IsEnabled && resource.BundleResourceContext != null;
+            bool isBundleParallelOperation =
+                _bundleOrchestrator.IsEnabled &&
+                resource.BundleResourceContext != null &&
+                resource.BundleResourceContext.IsParallelBundle;
 
-            if (isBundleOperation)
+            if (isBundleParallelOperation)
```

Comment:
I am not sure I understand this change.  Why do we check this only for parallel bundles if before it applied whether or not the bundle was a parallel bundle?

Reply:
Previously, BundleResourceContext was a concept only used by parallel bundles. 

The use of BundleResourceContext had to be expanded to support other parts of the service to identify if the requests were executed under the context of a bundle or not. 

My changes are expanding BundleResourceContext to all types of bundles (sequential and parallel), but it's now differentiated by the addition of the ProcessingLogic property to BundleResourceContext.

That's why I added the validation: to validate that the context is requesting the bundle to be executed in parallel, otherwise it should follow the previous logic and run sequentially.

Reply:
Sounds good!